### DISCUSSION TITLE
Wrapping is cancelled by executing commands via an active string but not while executing them normally (via the Define and Run Macro dialog)

### DISCUSSION THREAD

**Discussion by tozGitHub** [WinEdt-Team/WinEdt-Support#7] [1.2 Bugs and Issues]
Dear Alex, dear community,

I hope that I understood correctly that this category is for reporting bugs.

Suppose, we have the following file: (the orange dot marks the position of the cursor)
<img width="182" alt="Pasted image 20230315215316" src="https://github.com/WinEdt-Team/WinEdt-Support/assets/52357589/ae07f033-5f13-42c3-bbf9-cb088134d27e">
Suppose ActiveStrings.ini contains the following entry:
<img width="433" alt="Pasted image 20230315215549" src="https://github.com/WinEdt-Team/WinEdt-Support/assets/52357589/bbd4f13f-540f-4fb0-83bc-6ad4d9d768f5">
and the referred file Ed.edt looks like this:
![grafik](https://github.com/WinEdt-Team/WinEdt-Support/assets/52357589/dde26f9b-f43e-4110-a476-e6374ad27316)
Then activating the active string (with typing "ff  ") leads to:
<img width="299" alt="Pasted image 20230315215831" src="https://github.com/WinEdt-Team/WinEdt-Support/assets/52357589/ece6f14e-7a61-4969-8698-5b6bcaaa60d1">
The wrapping is cancelled.
While executing the macro:
BeginGroup;Backspace(4);Exe('%b\Navigation\Ed.edt');EndGroup
leads to the behaviour I would like to have and expect (the string "abcd" is deleted because I did not write the "ff ""-part):
<img width="132" alt="Pasted image 20230315220011" src="https://github.com/WinEdt-Team/WinEdt-Support/assets/52357589/0004b69d-4ec9-4794-b1bd-e56982f06dea">

[Files.zip](https://github.com/WinEdt-Team/WinEdt-Support/files/14104215/Files.zip) contains the TeX-file for testing (Test.tex), ActiveString.ini, and the mentioned macro file (Ed.edt).

WinEdt Build: 20230519  (v. 11.1) - 64-bit
Operating system: Microsoft Windows 11 Home (10.0.22631 Build 22631)

Best,
Thomas

**Comment by Alex-Simonic** [comment]
This is indeed an odd (unintended) behavior. It does not come as a surprise that triggering a macro code via an active string may act differently since the code is executed inside another event handler while processing character input.

After a character is typed a few things have to be checked in particular order: screen translations, delimiter matching, abbreviation, active string, auto-completion, spelling, and also auto-wrapping (if needed). Your active string macro response selects a few lines and paragraph wrapping assumes that you want to format selected lines into a single paragraph and proceed accordingly (resulting in a described behavior). This code was unchanged for over 15 years and it never resulted in problems because no active string response in the default settings changed the text selection as your macro does.

If you put SetSel(2); at the end of Ed.edt or disable wrapping then the "problem" will not happen. But that is probably not what you want to do.

I have now modified the algorithm so that it ignores the selection (for wrapping purposes) after executing the active string event. This seems to fix the problem without introducing additional oddities that I could detect: your example is the only one affected by it...

In other words: this is now fixed for the next version.
